src/dashboard/src/components/archival_storage/atom.py

```diff
@@ -0,0 +1,104 @@
+            extract_file(aip_uuid, mets_path, temp.name)
+
+            client = get_atom_client()
+            mw = METSDocument.fromfile(temp.name)
+
+            # Create file container
+            logger.info('Creating file container with slug %s and title %s', parent_slug, aip_name)
+            file_slug = client.add_child(parent_slug=parent_slug, title=aip_name, level='File')
+
+            def add_prop_from_xml(dict_, name, el, xpath):
+                """
+                Write to a dictionary a new pair with the given key and the value
+                taken from the text attribute of the element matched by the given
+                XPath query.
+                """
+                res = el.find(xpath)
+                if res is not None and res.text:
+                    dict_[name] = res.text
+                    logger.debug('Extracted property %s from METS: %s', name, res.text)
+                logger.debug('Failed to extract property %s from METS: not found', name)
+
+            # Add objects
+            for item in mw.all_files():
+                if item.type == 'Directory' or item.use != 'original':
+                    continue
+                attrs = {
+                    'title': os.path.basename(item.path),
+                    'usage': 'Offline',
+                    'file_uuid': item.file_uuid,
+                    'aip_uuid': aip_uuid,
+                }
+                amdsec = item.amdsecs[0].serialize()
+                add_prop_from_xml(attrs, 'size', amdsec, './/{info:lc/xmlns/premis-v2}objectCharacteristics/{info:lc/xmlns/premis-v2}size')
+                add_prop_from_xml(attrs, 'format_name', amdsec, './/{info:lc/xmlns/premis-v2}objectCharacteristics/{info:lc/xmlns/premis-v2}format/{info:lc/xmlns/premis-v2}formatDesignation/{info:lc/xmlns/premis-v2}formatName')
+                add_prop_from_xml(attrs, 'format_version', amdsec, './/{info:lc/xmlns/premis-v2}objectCharacteristics/{info:lc/xmlns/premis-v2}format/{info:lc/xmlns/premis-v2}formatDesignation/{info:lc/xmlns/premis-v2}formatVersion')
+                add_prop_from_xml(attrs, 'format_registry_name', amdsec, './/{info:lc/xmlns/premis-v2}objectCharacteristics/{info:lc/xmlns/premis-v2}format/{info:lc/xmlns/premis-v2}formatRegistry/{info:lc/xmlns/premis-v2}formatRegistryName')
+                add_prop_from_xml(attrs, 'format_registry_key', amdsec, './/{info:lc/xmlns/premis-v2}objectCharacteristics/{info:lc/xmlns/premis-v2}format/{info:lc/xmlns/premis-v2}formatRegistry/{info:lc/xmlns/premis-v2}formatRegistryKey')
+                title = os.path.basename(item.path)
+                slug = client.add_child(parent_slug=file_slug, title=title, level='Item')
+                logger.info('Creating child with slug %s and title %s', slug, title)
+                client.add_digital_object(slug, **attrs)
+
+            return file_slug
+
+    except (AtomError, MetsError, SlumberHttpBaseException):
```

Comment:
Can you move the exception handling to be around the smallest piece likely to generate it? It makes it clearer what the expected problem is, and means that unexpected errors don't get accidentally caught.